core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -18,12 +18,22 @@
 package org.apache.spark.storage
 
 import java.io.{InputStream, IOException}
+import java.net.URL
+import java.nio.ByteBuffer
 import java.util.concurrent.LinkedBlockingQueue
 
 import scala.collection.mutable.{ArrayBuffer, HashSet, Queue}
 import scala.util.{Failure, Success, Try}
 
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.crypto.CryptoCodec
+import org.apache.hadoop.crypto.CryptoInputStream
+import org.apache.hadoop.mapreduce.CryptoUtils
```

Comment:
These things are private and unstable.  We shouldn't be using private things from mapreduce/hadoop in spark.  It leads to issues when they change the api it breaks Spark.